eng/scripts/Cadl-Project-Sync.ps1

```diff
@@ -9,6 +9,19 @@ $ErrorActionPreference = "Stop"
 . $PSScriptRoot/../common/scripts/Helpers/PSModule-Helpers.ps1
 Install-ModuleIfNotInstalled "powershell-yaml" "0.4.1" | Import-Module
 $sparseCheckoutFile = ".git/info/sparse-checkout"
+$sdkGitRemoteAPI = "https://api.github.com/repos/Azure/azure-sdk-for-net/git/refs/heads/main"
+$sdkGitRemote = "https://github.com/Azure/azure-sdk-for-net.git"
+
+function GetLatestCommit() {
```

Comment:
Why do we need to do this? If we are just picking the latest then the clone will do that for us.

Reply:
The scenario is: when we want to build a new version of SDK, the new API depends on how the previous one looks like. E.g., https://github.com/Azure/autorest.csharp/issues/2887

Reply:
@m-nash I assume this is something you are helping design? I'm a little worried about us trying to pull source from another sha and download files from other commits in the middle of our generation processes.

Reply:
I think this code should not be part of the cadl scripts since its not unique to cadl.

@weshaggard in order to generate proper version overloads we need to know the previous contract.  We can use either the package managers (nuget, npm, etc) or we can sync to the release tag of the last GA version.  If we do the sync it can be reused by all languages, if we do the package managers its unique code for everyone.

We should definitely have this sync turned off by default and only enabled once a language generator needs it as not all will need this for back compatability right away.

I also think there is opportunity here for having these things pre-fetched to minimize download / network usage since that is already an issue in some of our pipelines today.

@pshao25  can we mark this as draft as we work through the prototype?